subsys/bluetooth/host/id.c

```diff
@@ -1273,12 +1273,11 @@ int bt_id_create(bt_addr_le_t *addr, uint8_t *irk)
 		}
 	}
 
-	new_id = bt_dev.id_count;
+	new_id = bt_dev.id_count++;
 	err = id_create(new_id, addr, irk);
 	if (err) {
+		bt_dev.id_count--;
```

Comment:
So `id_create()` calls a function that uses `bt_dev.id_count` instead of `new_id` @theob-pro ?

Reply:
`id_create()` will call `bt_settings_save_id()`, which will use the number of identity (`bt_dev.id_count`) to store the settings. `new_id` is the value of the new ID we want to create, the value is `bt_dev.id_count` before the creation of this new ID.
So it's not an error that `bt_settings_save_id()` use `bt_dev.id_count` instead of `new_id`.

Reply:
@carlescufi I explained it to myself like this: `id_create` takes a specific `new_id`. It is the responsibility of the caller to allocate an id number. `bt_dev.id_count++` is the allocation. `bt_dev.id_count--` is the corresponding de-allocation.